### PR TITLE
feat: Added manage corporate cards page and corporate card component

### PR DESCRIPTION
**Pull request by OmkarJ13**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=_0JBdeb)
### <samp>🤖 Generated by Copilot at 85c5a74</samp>

Added a new page and a new component for managing corporate cards in the app. The page is accessible from the fyle module and has a route, a template, a style, a spec, and a component file. The component is a shared component that will display the corporate card details and actions. The component also has a template, a spec, and a component file. The changes are part of the feature implementation for allowing the users to manage their corporate cards from the app.

### <samp>🤖 Generated by Copilot at 85c5a74</samp>

> _To manage their cards from the app_
> _The users will need a new tab_
> _So we added some files_
> _With components and styles_
> _And a route with `loadChildren` and `path`_

### <samp>🤖 Generated by Copilot at 85c5a74</samp>

*  Add a new feature for allowing the users to manage their corporate cards from the app ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-f2a0fea6976515e18b4e0c314651557c401be036adcbb955b4cde24dcb1d7169R1-R17), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-5e388261224399599ffa0e4acb99b0c7420e6c17d17c9185be65619fa93d7f31R1-R16), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-a9990626fe6e417ac0088e9e79ea18f1e0b64e1f644e8cead61870540da85a5eR1-R12), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-b7dfee1b6da49e0de48d7f654d6f1e4d479928ba336a24eb58e1331b0adc2be7R1-R15), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-b389a5d0241bda35125b36240b1d1cc470932c179782063153e1d1ad67705ed8R1-R24), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fR1-R15), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-1c08c3139af341f98f5dc83524b76783d3ef70c489281b15d00d2da63c1c0f39R1), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-393d21e7c8a24b907917f5b801dbd08fc2361a37bd2ea29d075aac4598fa406cR1-R24), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-18a2f0b572fcd52e51c9b964dc961c08ab88f1a626d76881dcdf1c42f05f1c3bR1-R8), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-c93d84613627aaadd08e7a7f1d5b20cde2542aacddaf14d50ed5a0676b733036L110-R190))
  - Create a new `manage-corporate-cards` page component under the `fyle` module, with its own routing, module, template, styles, and spec files ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-f2a0fea6976515e18b4e0c314651557c401be036adcbb955b4cde24dcb1d7169R1-R17), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-5e388261224399599ffa0e4acb99b0c7420e6c17d17c9185be65619fa93d7f31R1-R16), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-a9990626fe6e417ac0088e9e79ea18f1e0b64e1f644e8cead61870540da85a5eR1-R12), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-b7dfee1b6da49e0de48d7f654d6f1e4d479928ba336a24eb58e1331b0adc2be7R1-R15), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-b389a5d0241bda35125b36240b1d1cc470932c179782063153e1d1ad67705ed8R1-R24), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fR1-R15))
  - Add a new route for the `manage-corporate-cards` page in the `fyle-routing` module, as a child of the `fyle` route ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-c93d84613627aaadd08e7a7f1d5b20cde2542aacddaf14d50ed5a0676b733036L110-R190))
  - Import the `router` service in the `manage-corporate-cards` page component and define a `goBack` method that navigates to the `my_profile` page ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fR1-R15))
  - Display a header with a toolbar, a title, and a back button in the `manage-corporate-cards` page template ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-a9990626fe6e417ac0088e9e79ea18f1e0b64e1f644e8cead61870540da85a5eR1-R12))
  - Apply some SCSS rules for the toolbar and the title in the `manage-corporate-cards` page styles ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-b7dfee1b6da49e0de48d7f654d6f1e4d479928ba336a24eb58e1331b0adc2be7R1-R15))
  - Check if the `manage-corporate-cards` page component is created in the spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-b389a5d0241bda35125b36240b1d1cc470932c179782063153e1d1ad67705ed8R1-R24))
  - Create a new `corporate-card` component under the `shared` module, with its own template, spec, and component files ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-1c08c3139af341f98f5dc83524b76783d3ef70c489281b15d00d2da63c1c0f39R1), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-393d21e7c8a24b907917f5b801dbd08fc2361a37bd2ea29d075aac4598fa406cR1-R24), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-18a2f0b572fcd52e51c9b964dc961c08ab88f1a626d76881dcdf1c42f05f1c3bR1-R8))
  - Display a placeholder text in the `corporate-card` component template ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-1c08c3139af341f98f5dc83524b76783d3ef70c489281b15d00d2da63c1c0f39R1))
  - Check if the `corporate-card` component is created in the spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-393d21e7c8a24b907917f5b801dbd08fc2361a37bd2ea29d075aac4598fa406cR1-R24))
  - Leave the `corporate-card` component logic empty for now, to be updated with more content and functionality in the future ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-18a2f0b572fcd52e51c9b964dc961c08ab88f1a626d76881dcdf1c42f05f1c3bR1-R8))
* Add type annotations for the `loadChildren` functions in the `fyle-routing` module, to improve the readability, consistency, and type checking of the code ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-c93d84613627aaadd08e7a7f1d5b20cde2542aacddaf14d50ed5a0676b733036L4-R107), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-c93d84613627aaadd08e7a7f1d5b20cde2542aacddaf14d50ed5a0676b733036L72-R134), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-c93d84613627aaadd08e7a7f1d5b20cde2542aacddaf14d50ed5a0676b733036L95-R151), [link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-c93d84613627aaadd08e7a7f1d5b20cde2542aacddaf14d50ed5a0676b733036L110-R190))
  - Move the imports of the page modules to the top of the file, and use the `import().then()` syntax for the `loadChildren` functions ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-c93d84613627aaadd08e7a7f1d5b20cde2542aacddaf14d50ed5a0676b733036L4-R107))
  - Add the `Route['loadChildren']` type annotation for the `loadChildren` functions for the `notifications`, `my_create_report`, `add_edit_per_diem`, `add_edit_mileage`, and `add_edit_advance_request` routes ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-c93d84613627aaadd08e7a7f1d5b20cde2542aacddaf14d50ed5a0676b733036L72-R134))
  - Add the `Route['loadChildren']` type annotation for the `loadChildren` functions for the `split_expense`, `camera_overlay`, and `personal_cards_matched_expenses` routes ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-c93d84613627aaadd08e7a7f1d5b20cde2542aacddaf14d50ed5a0676b733036L95-R151))
  - Add the `Route['loadChildren']` type annotation for the `loadChildren` functions for the `personal_cards`, `view_expense`, `view_mileage`, `view_per_diem`, `potential_duplicates`, and `merge_expense` routes ([link](https://github.com/fylein/fyle-mobile-app/pull/2222/files?diff=unified&w=0#diff-c93d84613627aaadd08e7a7f1d5b20cde2542aacddaf14d50ed5a0676b733036L110-R190))

## Clickup
https://app.clickup.com/t/85ztm6dun

## Code Coverage
Please add code coverage here

## UI Preview
https://www.loom.com/share/73e1cb09250e46ef91dddd2ae8fbf8a4

### CODE REVIEW

File: src/app/fyle/fyle-routing.module.ts [hunk 1]
```diff
@@ -1,134 +1,192 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
 
+import { MyAdvancesPageModule } from './my-advances/my-advances.module';
```
Comment (OmkarJ13) [inline src/app/fyle/fyle-routing.module.ts:4]: Bunch of lint fixes here

File: src/app/fyle/manage-corporate-cards/manage-corporate-cards.page.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      <div class="manage-corporate-cards__cards-list">
+        <app-corporate-card
+          *ngFor="let corporateCard of corporateCards"
+          [card]="corporateCard"
+          (cardOptionsClick)="openCardOptions($event)"
+          [isVisaRTFEnabled]="isVisaRTFEnabled$ | async"
+          [isMastercardRTFEnabled]="isMastercardRTFEnabled$ | async"
+        ></app-corporate-card>
+      </div>
+    </ng-container>
```
Comment (OmkarJ13) [inline src/app/fyle/manage-corporate-cards/manage-corporate-cards.page.html:24]: Empty state and shimmers for the manage corporate card page will come in separate PR

File: src/app/fyle/manage-corporate-cards/manage-corporate-cards.page.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  prepareActionSheetButtons(card: PlatformCorporateCard): Observable<ActionSheetButton[]> {
+    return forkJoin([this.isVisaRTFEnabled$, this.isMastercardRTFEnabled$]).pipe(
+      map(([isVisaRTFEnabled, isMastercardRTFEnabled]) => {
+        const actionSheetButtons: ActionSheetButton[] = [];
+
+        if (card.is_visa_enrolled || card.is_mastercard_enrolled) {
+          actionSheetButtons.push({
+            text: 'Disconnect',
+            handler() {
+              // TODO: Disconnect
```
Comment (OmkarJ13) [inline src/app/fyle/manage-corporate-cards/manage-corporate-cards.page.ts:59]: API integration will be done in separate PR

File: src/app/fyle/manage-corporate-cards/manage-corporate-cards.page.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    </ion-buttons>
+    <ion-title class="manage-corporate-cards--title" mode="md">Manage Corporate Cards</ion-title>
+  </ion-toolbar>
+</ion-header>
+
+<ion-content>
+  <div class="manage-corporate-cards__cards">
+    <ng-container *ngIf="corporateCards$ | async as corporateCards">
+      <div class="manage-corporate-cards__cards-list">
+        <app-corporate-card
```
Comment (OmkarJ13) [inline src/app/fyle/manage-corporate-cards/manage-corporate-cards.page.html:16]: Empty state and shimmers for the manage corporate card page will come in separate PR

File: src/app/core/services/corporate-credit-card-expense.service.ts [hunk 5]
```diff
@@ -68,15 +80,15 @@ export class CorporateCreditCardExpenseService {
       );
   }
 
-  markPersonal(corporateCreditCardExpenseGroupId: string) {
+  markPersonal(corporateCreditCardExpenseGroupId: string): Observable<void> {
     return this.apiService.post('/corporate_credit_card_expenses/' + corporateCreditCardExpenseGroupId + '/personal');
   }
 
-  dismissCreditTransaction(corporateCreditCardExpenseId: string) {
+  dismissCreditTransaction(corporateCreditCardExpenseId: string): Observable<void> {
```
Comment (OmkarJ13) [inline src/app/core/services/corporate-credit-card-expense.service.ts:87]: These APIs don't return anything

File: src/app/core/enums/data-feed-source.enum.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+export enum DataFeedSource {
```
Comment (OmkarJ13) [inline src/app/core/enums/data-feed-source.enum.ts:1]: Figured out various data feed sources from backend APIs, and from the DB

File: src/app/fyle/manage-corporate-cards/corporate-card/corporate-card.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { IonicModule } from '@ionic/angular';
+
+import { CorporateCardComponent } from './corporate-card.component';
+
+xdescribe('CorporateCardComponent', () => {
```
Comment (OmkarJ13) [inline src/app/fyle/manage-corporate-cards/corporate-card/corporate-card.component.spec.ts:6]: Tests will come in next PR

File: src/app/core/services/corporate-credit-card-expense.service.ts [hunk 8]
```diff
@@ -68,15 +80,15 @@ export class CorporateCreditCardExpenseService {
       );
   }
 
-  markPersonal(corporateCreditCardExpenseGroupId: string) {
+  markPersonal(corporateCreditCardExpenseGroupId: string): Observable<null> {
     return this.apiService.post('/corporate_credit_card_expenses/' + corporateCreditCardExpenseGroupId + '/personal');
   }
 
-  dismissCreditTransaction(corporateCreditCardExpenseId: string) {
+  dismissCreditTransaction(corporateCreditCardExpenseId: string): Observable<null> {
```
Comment (OmkarJ13) [inline src/app/core/services/corporate-credit-card-expense.service.ts:87]: These APIs don't return anything

File: src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts [hunk 9]
```diff
@@ -600,7 +600,7 @@ export function TestCases1(getTestBed) {
       it('should mark a CCC txn as personal', (done) => {
         transactionService.delete.and.returnValue(of(expenseData1));
         trackingService.deleteExpense.and.returnValue(null);
-        corporateCreditCardExpenseService.markPersonal.and.returnValue(of(expenseData1));
+        corporateCreditCardExpenseService.markPersonal.and.returnValue(of(null));
```
Comment (OmkarJ13) [inline src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts:603]: Same as above, hence fixed the test files

File: src/assets/svg/mastercard-logo.svg [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<svg width="21" height="16" viewBox="0 0 21 16" fill="none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (Dimple16) [inline src/assets/svg/mastercard-logo.svg:1]: remove height and width from svg icons, we should be settings these via css

Comment (Dimple16) [reply src/assets/svg/mastercard-logo.svg:1]: same for all icons

File: src/app/shared/components/card-number/card-number.component.scss [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    display: flex;
+    gap: 4px;
+
+    &__dot {
+      border-radius: 50%;
+      padding: 2px;
+      background-color: $blue-black;
+    }
+
+    &__forth-dot {
```
Comment (Dimple16) [inline src/app/shared/components/card-number/card-number.component.scss:18]: Do you mean the `fourth-dot`? 😛

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

**Review by Dimple16 — APPROVED** [review]
added minor comments

File: src/app/fyle/manage-corporate-cards/corporate-card/corporate-card.component.scss [hunk 12]
```diff
@@ -0,0 +1,87 @@
+@import '../../../../theme/colors';
+
+.corporate-card {
+  width: 100%;
+  height: 162px;
```
[the remark this answers is not in the document]
Comment (OmkarJ13) [reply src/app/fyle/manage-corporate-cards/corporate-card/corporate-card.component.scss:5]: Had a small bug here where the background would get repeated for larger screen resolutions, I have fixed this. Thanks for letting know about responsiveness here 🙏

**Pull request merged**